lib/rucio/daemons/replicarecoverer/suspicious_replica_recoverer.py

```diff
@@ -338,49 +334,101 @@ def run_once(heartbeat_handler: Any, younger_than: int, nattempts: int, vos: Opt
-                                            break
-                                else:
-                                    logger(logging.WARNING, "RSE: %s, replica name %s, surl %s: Match for the metadata 'datatype' (%s) of replica found in json file, but no match for 'action' (%s)",
-                                           rse_key, replica_key, recoverable_replicas[vo][rse_key][replica_key]['surl'], i["datatype"], i["action"])
-                                break
-                            else:
-                                # If no policy has been set, default to ignoring the file (no action taken).
+                            # Check if a policy has been set for the specific datatype or scope
+                            # or if there are any wildcard policies regarding the scope.
+                            # Datatype action has priority over scope action.
+                            # Specific scope has priority over wildcard scope.
+                            if "datatype" in i:
+                                if file_metadata_datatype in i["datatype"]:
+                                    try:
+                                        action = i["datatype_action"]
+                                        in_file_metadata_datatype = True
+                                        break
+                                    except:
+                                        logger(logging.WARNING, "RSE: %s, replica name %s, surl %s: Datatype %s found in the policy library, but an error occured while retrieving the policy assigned to it. No action will be taken.",
+                                               rse_key, replica_key, recoverable_replicas[vo][rse_key][replica_key]['surl'], file_metadata_datatype)
+                                        break
+                            if ("scope" in i) and not in_file_metadata_datatype:
+                                # print("file_metadata_scope:", file_metadata_scope)
+                                # print("i[scope]:", i["scope"])
+                                # print("file_metadata_scope in i[scope]:", file_metadata_scope in i["scope"])
+                                if file_metadata_scope in i["scope"]:
+                                    try:
+                                        action = i["scope_action"]
+                                        in_file_scope = True
+                                    except:
+                                        logger(logging.WARNING, "RSE: %s, replica name %s, surl %s: Scope %s found in the policy library, but an error occured while retrieving the policy assigned to it. No action will be taken.",
+                                               rse_key, replica_key, recoverable_replicas[vo][rse_key][replica_key]['surl'], file_scope)
+                                        break
+                            if ("scope_wildcard" in i) and not (in_file_metadata_datatype or in_file_scope):
```

Comment:
Merge `scope_wildcard` with scope `scope

Reply:
The idea is that if a policy is set for a scope wildcard, but a single scope affected by that wildcard should have a different policy, then the policy for that single scope will take priority over the wildcard policy for that one scope..